device/src/drivers/ble/mesh/model/sensor/mod.rs

```diff
@@ -625,6 +630,26 @@ where
     pub fn new(data: C::Data<'a>) -> Self {
         Self { data }
     }
+
+    fn parse(parameters: &[u8]) -> Result<Self, ParseError> {
+        let mut pos = 0;
+        let mut result = None;
+        for d in C::DESCRIPTORS {
```

Comment:
If you look at the 'encode', the property descriptor can be either 2 or 3 bytes. The decoded value should also be verified that it's matching `d`.

Reply:
I think we're missing a bit logic in properly parsing PropertyID ... Ideally, I'd just do (pseudo code)

```
if PropertyId::parse(params) != d.id {
    return Err(ParseError::InvalidValue);
}
```

but currently `parse()` assumes two bytes and doesn't work for my use case.

I'll take this after the break in this or separate PR.

Reply:
@lulf this is now implemented. Property ids are properly parsed and checked.